### PR TITLE
Feature/UI utils

### PR DESCRIPTION
**Pull request by RagdollGoDown**
# UI Helper Functions
### Search Bar
We now have a search bar as shown in the Figma. It can be used like a normal TextField composable.

### Recipe List
A recipe listing as in the Figma. We're going to need a list similar to this one in multiple different screens, ex: search, created recipes, etc.

### PlateSwipe Scaffold
Probably the most important, a scaffold that has the PlateSwipe theme, that can be used on our screen. When putting content inside of the scaffold, make sure to put it in the content parameter. It's important we have this now so that we avoid repeating code between screens and preserve a the same theme throughout the app. It also simplifies our job if we need to change that them afterwards.

image

Here we can see all three put in a sample screen. The images haven't loaded, but the logic for showing the images is the same so it should show up in the build.

### Minor Additions
- titleMedium typography in theme
- fix bottom navigation menu duplicate test tags
- remove unused recipe screen tests
- removed unnecessary code in account screen

## Checklist
- [x] Tests added.
- [x] Documentation updated.
- [x] All CI checks passed.
- [x] Linked issue/feature (if applicable): #134

### CODE REVIEW

**Review by AdriB02 — COMMENTED** [review]
I noticed that the RecipeScreenTest has been removed. Could you please clarify the reason for its deletion? Overall, your work is impressive, and the line coverage is excellent! I really appreciate the effort you've put into this. However, I recommend adding comments throughout the code to enhance readability and make it easier for others to understand your thought process.

Great job overall, and thanks for your work on this!

File: app/src/main/java/com/android/sample/ui/utils/RecipeList.kt [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
+import com.android.sample.model.recipe.Recipe
+import com.android.sample.ui.theme.goldenBronze
+
+val cornerEdgeRadius: Dp = 12.dp
+
```
[the remark this answers is not in the document]
Comment (RagdollGoDown) [reply app/src/main/java/com/android/sample/ui/utils/RecipeList.kt:49]: Will do!

File: app/src/main/java/com/android/sample/ui/utils/RecipeList.kt [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+                      RecipeRating()
+                    }
+
+                Row(
+                    modifier = Modifier.fillMaxWidth().weight(1f),
+                    horizontalArrangement = Arrangement.SpaceBetween) {
+                      RecipeCategories(recipe)
+
+                      RecipePrice(cost = 3, recipe = recipe)
+                    }
```
[the remark this answers is not in the document]
Comment (RagdollGoDown) [reply app/src/main/java/com/android/sample/ui/utils/RecipeList.kt:116]: It's simply a placeholder for now

File: app/src/main/java/com/android/sample/ui/utils/RecipeList.kt [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+              }
+        }
+  }
+}
+
+@Composable
+private fun RecipePrice(maxDollars: Int = 3, cost: Int, recipe: Recipe) {
+  Row(
```
[the remark this answers is not in the document]
Comment (RagdollGoDown) [reply app/src/main/java/com/android/sample/ui/utils/RecipeList.kt:124]: We're copying the three dollar signs from google maps

File: app/src/main/java/com/android/sample/ui/utils/RecipeList.kt [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+      fontWeight = FontWeight.Bold,
+  )
+}
+
+@Composable
+private fun RecipeLike(recipe: Recipe) {
+  var isLiked by remember { mutableStateOf(true) }
+  Icon(
+      imageVector = if (isLiked) Icons.Filled.Favorite else Icons.Filled.FavoriteBorder,
+      contentDescription = "like",
```
[the remark this answers is not in the document]
Comment (RagdollGoDown) [reply app/src/main/java/com/android/sample/ui/utils/RecipeList.kt:207]: I'll do so

File: app/src/main/java/com/android/sample/ui/utils/RecipeList.kt [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+@Composable
+private fun RecipePrice(maxDollars: Int = 3, cost: Int, recipe: Recipe) {
+  Row(
+      verticalAlignment = Alignment.CenterVertically,
+      modifier = Modifier.testTag("priceRating${recipe.idMeal}")) {
+        for (i in 1..maxDollars) {
+          val isSelected = i <= cost
+          val icon = Icons.Filled.AttachMoney
+          val iconTintColor = if (isSelected) Color(0xFF000000) else Color(0xFFB0B0B0)
```
[the remark this answers is not in the document]
Comment (RagdollGoDown) [reply app/src/main/java/com/android/sample/ui/utils/RecipeList.kt]: In one of the last commits, I put it in the color file

File: app/src/main/java/com/android/sample/ui/utils/RecipeList.kt [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+@Composable
+private fun RecipePrice(maxDollars: Int = 3, cost: Int, recipe: Recipe) {
+  Row(
+      verticalAlignment = Alignment.CenterVertically,
+      modifier = Modifier.testTag("priceRating${recipe.idMeal}")) {
+        for (i in 1..maxDollars) {
+          val isSelected = i <= cost
+          val icon = Icons.Filled.AttachMoney
```
[the remark this answers is not in the document]
Comment (RagdollGoDown) [reply app/src/main/java/com/android/sample/ui/utils/RecipeList.kt]: I installed a local Icon

File: app/src/main/java/com/android/sample/ui/utils/RecipeList.kt [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+  var isLiked by remember { mutableStateOf(true) }
+  Icon(
+      imageVector = if (isLiked) Icons.Filled.Favorite else Icons.Filled.FavoriteBorder,
+      contentDescription = "like",
+      modifier =
+          Modifier.padding(4.dp).testTag("recipeFavoriteIcon${recipe.idMeal}").clickable {
+            isLiked = !isLiked
+          },
+      tint = Color.Red)
+}
```
[the remark this answers is not in the document]
Comment (RagdollGoDown) [reply app/src/main/java/com/android/sample/ui/utils/RecipeList.kt:226]: I put a color from the figma

File: app/src/main/java/com/android/sample/ui/utils/SearchBar.kt [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+ *
+ * @param modifier the modifier to apply to this layout node.
+ * @param onValueChange the callback to invoke when the search text changes.
+ */
+@Preview
+@Composable
+fun SearchBar(modifier: Modifier = Modifier, onValueChange: (String) -> Unit = {}) {
+  var searchText by remember { mutableStateOf("") }
+
+  val cornerEdgeRadius: Dp = 16.dp
```
[the remark this answers is not in the document]
Comment (RagdollGoDown) [reply app/src/main/java/com/android/sample/ui/utils/SearchBar.kt]: Done!

File: app/src/main/java/com/android/sample/ui/utils/RecipeList.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (AdriB02) [reply app/src/main/java/com/android/sample/ui/utils/RecipeList.kt:125]: Ok thanks for your answer !

**Review by AdriB02 — APPROVED** [review]
Great work on these changes! The updates look solid, and your scaffold structure is really impressive. Could you update the project wiki to document the new scaffold details? I accept the PR. Thanks for the effort!

**Pull request merged**